prog.go

```diff
@@ -242,14 +242,24 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions) (*Program, er
 	insns := make(asm.Instructions, len(spec.Instructions))
 	copy(insns, spec.Instructions)
 
-	handle, fib, lib, err := btf.MarshalExtInfos(insns)
-	if err != nil && !errors.Is(err, btf.ErrNotSupported) {
-		return nil, fmt.Errorf("load ext_infos: %w", err)
+	var b btf.Builder
+	if err := applyRelocations(insns, opts.KernelTypes, spec.ByteOrder, &b); err != nil {
+		return nil, fmt.Errorf("apply CO-RE relocations: %w", err)
 	}
-	if handle != nil {
-		defer handle.Close()
 
-		attr.ProgBtfFd = uint32(handle.FD())
+	if !b.Empty() && haveProgramExtInfos() != nil {
+		// Return ErrNotSupported instead of E2BIG if CO-RE relies on BTF
```

Comment:
It's not clear what you mean by this. Doesn't CO-RE always rely on BTF being present?

Reply:
I'll rephrase. The issue is that a CO-RE local type id relocation might assert that a specific type exists in the BTF blob.